game-core/src/main/java/games/strategy/debug/ConsoleHandler.java

```diff
@@ -1,106 +1,53 @@
 package games.strategy.debug;
 
-import java.io.PrintStream;
-import java.io.UnsupportedEncodingException;
-import java.util.Optional;
-import java.util.function.Supplier;
-import java.util.logging.ErrorManager;
+import java.util.function.Consumer;
 import java.util.logging.Handler;
-import java.util.logging.Level;
 import java.util.logging.LogManager;
 import java.util.logging.LogRecord;
-import java.util.logging.SimpleFormatter;
 
-import com.google.common.annotations.VisibleForTesting;
+import lombok.AllArgsConstructor;
+import lombok.extern.java.Log;
 
 /**
- * A {@link Handler} that publishes log records of level {@link Level#WARNING} or {@link Level#SEVERE} to
- * {@code System.err} and log records of all other levels to {@code System.out}.
- *
+ * A {@link Handler} that publishes log records to an injected log handler. This can be wired up to send log messages
+ * to a UI console.
  * <p>
  * Configuration: This handler does not currently support configuration through the {@link LogManager}.
  * It always uses the following default configuration:
  * </p>
- *
  * <ul>
  * <li>Level: {@code Level.ALL}</li>
  * <li>Filter: No {@code Filter}</li>
  * <li>Formatter: {@code java.util.logging.SimpleFormatter}</li>
  * <li>Encoding: default platform encoding</li>
  * </ul>
  */
+@Log
+@AllArgsConstructor
 public final class ConsoleHandler extends Handler {
-  private final Supplier<PrintStream> errSupplier;
-  private final Supplier<PrintStream> outSupplier;
 
-  @SuppressWarnings("unused")
-  public ConsoleHandler() {
-    this(() -> System.out, () -> System.err);
-  }
-
-  @VisibleForTesting
-  ConsoleHandler(final Supplier<PrintStream> outSupplier, final Supplier<PrintStream> errSupplier) {
-    this.errSupplier = errSupplier;
-    this.outSupplier = outSupplier;
-    configure();
-  }
-
-  private void configure() {
-    setLevel(Level.ALL);
-    setFilter(null);
-    setFormatter(new SimpleFormatter());
-    try {
-      setEncoding(null);
-    } catch (final UnsupportedEncodingException e) {
-      throw new AssertionError("default platform encoding should always be available", e);
-    }
-  }
+  private final Consumer<LogRecord> msgReceiver;
 
   @Override
   public void close() {
-    flush();
```

Comment:
flushing output streams was probably not necessarily to begin with, but maybe it was. Regardless, there is not an equivalent for flushing to a text area, and even there is, considering it's done as part of shutdown, rendering content to a window that is closing/closed does not make sense. So these becomes no-ops (to boot, the proper cleanup is going to be done in the injected code, not here, this class is really just scaffolding at this point).